singer/catalog.py

```diff
@@ -1,10 +1,39 @@
 '''Provides an object model for a Singer Catalog.'''
+from . import metadata
 
 import json
 import sys
 
+from singer.bookmarks import get_currently_syncing
+from singer.logger import get_logger
 from singer.schema import Schema
 
+LOGGER = get_logger()
+
+def _shuffle_streams(catalog, state):
+    currently_syncing = get_currently_syncing(state)
+
+    if currently_syncing is None:
+        return catalog.streams
+
+    matching_index = 0
+    for i, catalog_entry in enumerate(catalog.streams):
+        if catalog_entry.tap_stream_id == currently_syncing:
+            matching_index = i
```

Comment:
You can put a break here once the index is found. Could be a bit of time, depending on how many streams there are (e.g., big database)